### PR TITLE
Remove ert config from batch sim

### LINKED ISSUE #9002: Remove ert config from BatchSimulator
This needs a subset of the config, so we should try to extract only the needed pars to make it more explicit.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #9002 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/cli/workflow.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def execute_workflow(
         msg = "Workflow {} is not in the list of available workflows"
         logger.error(msg.format(workflow_name))
         return
-    runner = WorkflowRunner(workflow, storage, ert_config=ert_config)
+    runner = WorkflowRunner(workflow=workflow, storage=storage, ert_config=ert_config)
```
Comment (yngve-sk) [inline src/ert/cli/workflow.py:23]: OK change, but purely cosmetic and unrelated to this PR? Could be in a separate commit

**Review by yngve-sk — COMMENTED** [review]
HMM, I see the number of parameters passed from the ert config  became bit large, a lot of it is from having to make `create_run_path` generic so it can be invoked from both places. And we do need to get the info from the ErtConfig somehow but the way it is currently structured is making this very verbose. It might be worth thinking a bit more about whether we need to use the same `create_run_path` from everest as from ERT, I'm not entirely sure..

I think this deserves some more discussion at least. Would it maybe be possible to make BatchSimulator depend more directly on things from the EverestConfig instead sub-configs of the ErtConfig?

File: src/ert/config/ert_config.py [hunk 2]
```diff
@@ -83,6 +83,164 @@ def site_config_location() -> str:
     )
 
 
+def create_forward_model_json(
```
[the remark this answers is not in the document]
Comment (frode-aarstad) [reply src/ert/config/ert_config.py:86]: Yes

**Comment by frode-aarstad** [comment]
> HMM, I see the number of parameters passed from the ert config became bit large, a lot of it is from having to make `create_run_path` generic so it can be invoked from both places. And we do need to get the info from the ErtConfig somehow but the way it is currently structured is making this very verbose. It might be worth thinking a bit more about whether we need to use the same `create_run_path` from everest as from ERT, I'm not entirely sure..
> 
> I think this deserves some more discussion at least. Would it maybe be possible to make BatchSimulator depend more directly on things from the EverestConfig instead sub-configs of the ErtConfig?

I agree that some functions have become very verbose. In particular BatchSimulator and BatchContext. Perhaps its worth having a discussion how we can improve those two.

**Review by yngve-sk — APPROVED** [review]
👍
